### ISSUE TITLE
Upload: lack of documentation (or support) for passing errors from `:external` 2-arity function

### ISSUE THREAD

**Issue by DaTrader** [phoenixframework/phoenix_live_view#3369]
### Environment

* Phoenix version (mix deps): 0.20.17

### Actual behavior

Upload config: external, autoload

I couldn't find any documentation on what the `:external` 2-arity function is supposed to return in the meta map in `{ :error, meta, socket}` for the error to end up in the `UploadConfig` structure i.e. to become retrievable with `Phoenix.Component.upload_errors`.

Has this feature been developed yet or is planned to be developed or it's completely up to developers to handle such errors separately from the predefined upload ones?

### Expected behavior

There should be a word in the docs on how to deal with those errors now and in the future.

PS. It would also be great if the value provided to `entry.error/1` in a JS uploader was actually propagated along or instead of the `external_client_failure` atom.

Thanks

**Comment by SteffenDE** [comment]
@DaTrader you're right, there was actually a bug that meant that for auto_upload the error was not reported. See https://github.com/phoenixframework/phoenix_live_view/pull/4397.
